src/winmenu.c

```diff
@@ -71,14 +71,19 @@ Menu *CreateWindowMenu(ClientNode *np)
       AddWindowMenuItem(menu, _("Minimize"), MA_MINIMIZE, np, 0);
    }
 
-   if(!(np->state.status & STAT_FULLSCREEN)) {
+   if(!(np->state.status & STAT_FULLSCREEN) && 
+      !(np->state.status & STAT_MINIMIZED)  && 
+       (np->state.border & BORDER_MIN))
+   {
```

Comment:
My opinion has changed ;)  You can not run commands:
- Shade
- Unshade
- Restore from maximizing